lab_stack/ConsoleApplication33/ConsoleApplication33/Stack.cpp

```diff
@@ -57,69 +56,69 @@
         }
     }
 
-    //�������� �����
+    //î÷èùåíèå ñòåêà
     void Stack::Clear()
     {
         Stack::~Stack();
         top = nullptr;
-        std::cout << "\x1b[32m���� ������ :)\x1b[0m" << std::endl;
+        std::cout << "\x1b[32mÑòåê î÷èùåí :)\x1b[0m" << std::endl;
     }
 
-    //�������� �������� ������������� ����������
+    //óäàëåíèå ýëåìåíòà îïðåäåëåííîãî ïðèîðèòåòà
     void Stack::DeleteByPriority(const int priority)
     {
-        Element* previous = top;//������ ���������� �������
-        Element* current = top;//������� �� ������� �������
+        Element* previous = top;//ññûëêà ïðåäûäóùèé ýëåìåíò
+        Element* current = top;//ñññûëêà íà òåêóùèé ýëåìåíò
 
-        if (current->priority == priority && current->next == nullptr) //������ ���� ������� � �����
+        if (current->priority == priority && current->next == nullptr) //òîëüêî îäèí ýëåìåíò â ñòåêå
         {
             top = nullptr;
         }
         else
         {
-            while (current != nullptr && current->priority != priority)//���������� �������� �����
+            while (current != nullptr && current->priority != priority)//ïåðåáèðàåì ýëåìåíòû ñòåêà
             {
                 previous = current;
                 current = current->next;
             }
 
-            if (current->priority == priority && current == top && previous == top )//������ ������� � ������ ����������� � ������ �����
+            if (current->priority == priority && current == top && previous == top )//íàéäåí ýëåìåíò ñ íóæíûì ïðèîðèòåòîì â íà÷àëå ñòåêà
             {
                 top = current->next;
                 delete current;
             }
 

```

Comment:
Can all these conditions be fulfilled simultaneously? Do we need to go into all the conditions all the time?